sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.aest.controller.ts

```diff
@@ -74,20 +76,9 @@ export class RestrictionAESTController extends BaseController {
   async getStudentRestrictions(
     @Param("studentId", ParseIntPipe) studentId: number,
   ): Promise<RestrictionSummaryAPIOutDTO[]> {
-    const studentRestrictions =
-      await this.studentRestrictionService.getStudentRestrictionsById(
-        studentId,
-      );
-    return studentRestrictions?.map((studentRestriction) => ({
-      restrictionId: studentRestriction.id,
-      restrictionType: studentRestriction.restriction.restrictionType,
-      restrictionCategory: studentRestriction.restriction.restrictionCategory,
-      restrictionCode: studentRestriction.restriction.restrictionCode,
-      description: studentRestriction.restriction.description,
-      createdAt: studentRestriction.createdAt,
-      updatedAt: studentRestriction.updatedAt,
-      isActive: studentRestriction.isActive,
-    }));
+    return this.restrictionControllerService.getStudentRestrictions(studentId, {
```

Comment:
To avoid the cast in the controllers a possible solution would be creating methods overloads where specific parameters would return specific types, as shown below.
```ts
async getStudentRestrictions(
  studentId: number,
  options: {
    extendedDetails?: boolean;
  },
): Promise<RestrictionSummaryAPIOutDTO[]>;
async getStudentRestrictions(
  studentId: number,
  options?: {
    filterNoEffectRestrictions?: boolean;
  },
): Promise<RestrictionInstitutionSummaryAPIOutDTO[]>;
async getStudentRestrictions(
  studentId: number,
  options?: {
    extendedDetails?: boolean;
    filterNoEffectRestrictions?: boolean;
    onlyActive?: boolean;
  },
): Promise<
  RestrictionInstitutionSummaryAPIOutDTO[] | RestrictionSummaryAPIOutDTO[]
> {
  const studentRestrictions =
    await this.studentRestrictionService.getStudentRestrictionsById(
      studentId,
      {
        filterNoEffectRestrictions: options?.filterNoEffectRestrictions,
        onlyActive: options?.onlyActive,
      },
    );
  return studentRestrictions?.map((studentRestriction) => ({
    restrictionId: studentRestriction.id,
    restrictionType: studentRestriction.restriction.restrictionType,
    restrictionCategory: studentRestriction.restriction.restrictionCategory,
    restrictionCode: studentRestriction.restriction.restrictionCode,
    description: studentRestriction.restriction.description,
    createdAt: studentRestriction.createdAt,
    updatedAt: options?.extendedDetails
      ? studentRestriction.updatedAt
      : undefined,
    isActive: studentRestriction.isActive,
  }));
}
```

Reply:
Looks good to me. I didn't know that is possible to do in TS :)